library/core/src/str/pattern.rs

```diff
@@ -601,6 +608,17 @@ where
     }
 }
 
+impl<const N: usize> MultiCharEq for &[char; N] {
+    #[inline]
+    fn matches(&mut self, c: char) -> bool {
+        match N {
+            0 => false,
+            1 => self[0] == c,
+            _ => self.iter().any(|&m| m == c),
+        }
```

Comment:
Actually I think I should leave the N optimization to the compiler...

Reply:
That makes sense to me -- especially since `.find(&[])` feels rather unlikely anyway.